jocd/src/br/org/certi/jocd/dapaccess/Command.java

```diff
@@ -355,9 +418,17 @@ public void add(int count, byte request, long[] words, Byte dapIndex) {
   /*
    * Decode the response data
    */
-  public byte[] decodeData(byte[] data) throws TransferError {
-    assert this.getEmpty() == false;
-    assert this.dataEncoded == true;
+  public byte[] decodeData(byte[] data) throws TransferError, Error {
+    // Assert this.getEmpty() == false.
+    if (this.getEmpty()) {
+      throw new Error("decodeData: Unexpected getEmpty() value ( " + getEmpty() + ")");
```

Comment:
This getEmpty() call could be avoided here, since the error state already occurred and it's already known that the return value will be true.

Reply:
Fixed on 3905d3fb99c8c6b556bfe3b84b6ed06c2f55747c